sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch/extensions/DateExtensions.kt

```diff
@@ -47,16 +49,40 @@ fun LocalDateTime.startOfNextDay(): LocalDateTime {
             .startOfDay()
 }
 
+fun LocalDateTime.inSameDayAs(date: LocalDateTime): Boolean {
+    return this.isBetweenInclusive(date.startOfDay(), date.endOfDay())
```

Comment:
Nice, and I do agree with you.  But, I did it this way because MySQL's BETWEEN statement is inclusive for both start and end, and since these extensions work in conjunction with schedule queries, I thought it was best to keep parity with that.